src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -566,10 +638,74 @@ void HMM<Distribution>::Smooth(const arma::mat& dataSeq,
   // Compute expected emissions.
   // Will not work for distributions without a Mean() function.
   smoothSeq.zeros(dimensionality, dataSeq.n_cols);
-  for (size_t i = 0; i < emission.size(); ++i)
+  for (size_t i = 0; i < emission.size(); i++)
     smoothSeq += emission[i].Mean() * exp(stateLogProb.row(i));
 }
 
+/**
+ * The Forward procedure (part of the Forward-Backward algorithm).
+ */
+template<typename Distribution>
+arma::vec HMM<Distribution>::ForwardAtT0(const arma::vec& emissionLogProb,
+                                double& logScales) const
+{
+  // Our goal is to calculate the forward probabilities:
+  //  P(X_k | o_{1:k}) for all possible states X_k, for each time point k.
+
+
+    ConvertToLogSpace();
+
+    arma::vec forwardLogProb(logTransition.n_rows);
+    forwardLogProb.fill(-std::numeric_limits<double>::infinity());
+    // The first entry in the forward algorithm uses the initial state
+    // probabilities.  Note that MATLAB assumes that the starting state (at
+    // t = -1) is state 0; this is not our assumption here.  To force that
+    // behavior, you could append a single starting state to every single data
+    // sequence and that should produce results in line with MATLAB.
+    for (size_t state = 0; state < logTransition.n_rows; state++) {
+        forwardLogProb(state) = logInitial(state) + emissionLogProb(state);
+    }
+
+    // Normalize probability.
+    logScales = math::AccuLog(forwardLogProb);
+    if (std::isfinite(logScales)){
+        forwardLogProb -= logScales;
+    }
```

Comment:
```suggestion
  if (std::isfinite(logScales))
   forwardLogProb -= logScales;
```
No need for braces for a one-liner if statement. :+1: